apf/jack_policy.h

```diff
@@ -219,9 +227,11 @@ jack_policy::Xput<X>::_init_port(const parameter_map& p, jack_policy& parent)
 
     name = p.get(X::prefix_name(), X::default_prefix()) + id;
   }
-
-  return X::is_input
-    ? parent.register_in_port(name) : parent.register_out_port(name);
+  JackClient::port_t* rport = X::is_input ? parent.register_in_port(name) : parent.register_out_port(name);
+  if (rport==NULL){
+    throw std::runtime_error("Could not register JACK ports!");
```

Comment:
```suggestion
    throw std::runtime_error("Could not register JACK port!");
```

It's only a single port, right?